libtenzir/include/tenzir/tql2/plugin.hpp

```diff
@@ -171,6 +168,190 @@ class parser_adapter final : public crtp_operator<parser_adapter<Parser>> {
   Parser parser_;
 };
 
+template <class Loader, detail::string_literal NameOverride = "">
+class loader_adapter final : public crtp_operator<loader_adapter<Loader>> {
```

Comment:
If you look at TQL1 `load_operator`, you'll see a couple more overrides:
- `detached() -> true`
- `internal() -> dispatch to loader`
- `location() -> local`